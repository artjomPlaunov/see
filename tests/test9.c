int test(int a, int b) {
  int c = 0;
  return c;
} 

int main() {
  int a[5] = {1,2,3,4,5};
  test(a[2], 100);
  return 0;
}
