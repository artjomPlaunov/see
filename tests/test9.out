(Ast.Prog
   [(Ast.Fun (Ast.INT_T,
       (Ast.ParamLst
          [(Ast.Param (Ast.INT_T, (Ast.Ident "a")));
            (Ast.Param (Ast.INT_T, (Ast.Ident "b")))]),
       (Ast.Ident "test"),
       (Ast.Block
          [(Ast.Stm_declLst (Ast.INT_T, [(Ast.Decl ((Ast.Ident "c"), 0))]));
            (Ast.Stm_exp (Ast.Return (Ast.Variable (Ast.Ident "c"))))])
       ));
     (Ast.Fun (Ast.INT_T, (Ast.ParamLst []), (Ast.Ident "main"),
        (Ast.Block
           [(Ast.Stm_arrayDecl ((Ast.Ident "a"), 5, [1; 2; 3; 4; 5]));
             (Ast.Stm_exp
                (Ast.FunctionCall ((Ast.Ident "test"),
                   [(Ast.ArraySubscript ((Ast.Ident "a"), 2));
                     (Ast.Constant 100)]
                   )));
             (Ast.Stm_exp (Ast.Return (Ast.Constant 0)))])
        ))
     ])